outer.c

```diff
@@ -315,6 +324,21 @@ void outer_helper_spawn(struct outer_helper *helper)
 
 	if (helper->cgroup_enabled) {
 		int cgroupfd = recv_fd(fdpair[SOCKET_CHILD]);
+
+		char *subcgroup = makepath("bst.%d", rootpid);
+
+		// Writing to cgroup directory likely requires privilige
+		make_capable(BST_CAP_DAC_OVERRIDE);
+
+		if (mkdirat(cgroupfd, subcgroup, 0777) == -1) {
+			err(1, "unable to create sub-hierachy cgroup %s", subcgroup);
+		}
+
+		reset_capabilities();
+
+		// Notify root process cgroup hierarchy has been created
+		send_fd(fdpair[SOCKET_CHILD], cgroupfd);
```

Comment:
Rather than doing this back-and-forth, I'd argue that the privileged helper could also be the one adding the process to cgroups.procs. It's just a matter of writing `child_pid` a handful of lines below rather than `"0"` to the file.

Reply:
This choice of having the root process write to the cgroup.procs was so that if it fails to write "0" to cgroup.procs the outer helper would still cleanup the created bst.<pid> file. It seemed like this could be a pretty common occurrence on a mis-configured system. Is this not really a concern?